landoscript/src/landoscript/script.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import os.path
+
+import aiohttp
+import scriptworker.client
+
+from landoscript import lando
+from landoscript.actions import version_bump
+from landoscript.github import GithubClient
+from scriptworker_client.github import extract_github_repo_owner_and_name
+
+log = logging.getLogger(__name__)
+
+
+def get_default_config(base_dir="") -> dict:
+    base_dir = base_dir or os.path.dirname(os.getcwd())
+    default_config = {
+        "work_dir": os.path.join(base_dir, "work_dir"),
+        "artifact_dir": os.path.join(base_dir, "artifact_dir"),
+        "schema_file": os.path.join(os.path.dirname(__file__), "data", "landoscript_task_schema.json"),
+    }
+    return default_config
+
+
+async def async_main(context):
+    config = context.config
+    payload = context.task["payload"]
+    artifact_dir = config["artifact_dir"]
+    public_artifact_dir = os.path.join(artifact_dir, "public", "build")
+    owner, repo = extract_github_repo_owner_and_name(payload["source_repo"])
```

Comment:
I think this will need to be extracted out of the scopes instead of the payload. Or else we'll need to validate scopes independently (but then `source_repo` would need to be defined in the task definition twice)

Reply:
Ah, indeed! I have a "add scopes" todo item still; I'll make sure to address this specific point as part of that.

Reply:
I pushed an update that deals with this and the next point. I ended up adding separate `repo` and `branch` scopes (this will be needed for the main Firefox repository, where we'll have separate branches in it for various L3 repos), as well as `action` scopes. If you have any concerns please let me know!